connect/runtime/src/test/java/org/apache/kafka/connect/runtime/rest/RestServerTest.java

```diff
@@ -64,17 +70,34 @@ public class RestServerTest {
     private Herder herder;
     private Plugins plugins;
     private RestServer server;
+    private CloseableHttpClient httpClient;
+    private Collection<CloseableHttpResponse> responses = new ArrayList<>();
 
     protected static final String KAFKA_CLUSTER_ID = "Xbafgnagvar";
 
     @Before
     public void setUp() {
         herder = mock(Herder.class);
         plugins = mock(Plugins.class);
+        httpClient = HttpClients.createMinimal();
     }
 
     @After
     public void tearDown() {
+        responses.stream().forEach(response -> {
+            try {
+                response.close();
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
```

Comment:
Any reason we shouldn't fail the test if this fails? It may indicate an issue with the server if we can't close a response or a client.

Reply:
A bit more complicated to correctly close everything but still raising errors.
Or we could just fail everything on first closing issue?

Added commit for 1st option. Can go for a fail-fast option if you prefer.

Reply:
I think fail-fast is fine. It may cause resource leaks during test runs, but if we do our due diligence and make sure tests pass before merging changes, this should never happen on trunk or backport branches, and should only affect local runs while iterating on changes.

Reply:
Done!